src/Acuminator/Acuminator.Analyzers/StaticAnalysis/LegacyBqlField/LegacyBqlFieldFix.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Acuminator.Analyzers.StaticAnalysis.LegacyBqlField
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class LegacyBqlFieldFix : CodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(Descriptors.PX1060_LegacyBqlField.Id);
+
+		public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			var classNode = root.FindNode(context.Span).FirstAncestorOrSelf<ClassDeclarationSyntax>();
+			if (classNode == null)
+				return;
+
+			var diagnostic = context.Diagnostics.FirstOrDefault(d => d.Id == Descriptors.PX1060_LegacyBqlField.Id);
+			if (diagnostic == null)
+				return;
+
+			if (diagnostic.Properties == null
+				|| !diagnostic.Properties.TryGetValue(LegacyBqlFieldAnalyzer.CorrespondingPropertyType, out string typeName)
+				|| String.IsNullOrEmpty(typeName))
+			{
+				return;
+			}
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			string title = nameof(Resources.PX1060Fix).GetLocalized().ToString();
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					title,
+					c => Task.FromResult(
+						context.Document.WithSyntaxRoot(
+							root.ReplaceNode(
+								classNode.BaseList,
+								BaseList(
+									SeparatedList(
+										new BaseTypeSyntax[]
+										{
+											SimpleBaseType(CreateBaseType(typeName, classNode.Identifier.Text))
+										}))))),
+					title),
+				context.Diagnostics);
+		}
+
+		private TypeSyntax CreateBaseType(string typeName, string dacFieldName)
+		{
+			if (!PropertyTypeToFieldType.ContainsKey(typeName))
+				throw new NotSupportedException();
+
+			return IdentifierName($"PX.Data.BQL.Bql{PropertyTypeToFieldType[typeName]}.Field<{dacFieldName}>");
+		}
+
+		public static readonly IReadOnlyDictionary<string, string> PropertyTypeToFieldType = new Dictionary<string, string>
```

Comment:
@Derrekam this register currently doesn't support string arrays, UInt16 arrays and float. We have such attributes and DAC properties - see FieldTypeAttributesRegister.cs class.

Reply:
there is no representation for such type of fields in FBQL, therefore there is no need to provide such support in acuminator

Reply:
Ok, but you need to make sure that the exception won't be thrown for such types. I suggest to add such fields to DAC in source files for unit tests